### ISSUE TITLE
[Homematic] hmip-ps no channels available

### ISSUE THREAD

**Issue by agroezinger** [eclipse/smarthome#6887]
Hey,

looks like the HmIP-PS is not supported yet. At least no channels are shown for this kind of device. Picture and more detailed description here: https://community.openhab.org/t/integrate-hmip-ps-not-working/59183

Kind regards

**Comment by MHerbst** [comment]
The binding has moved (back) to the openhab2-addons repo (https://github.com/openhab/openhab2-addons). Can you create the issue in this repo and close the issue here.

Please add the following information:
- your openHAB version
- you version of the CCU (or Homegear or whatever you use)

If would also help if you could add a trace log.

**Comment by agroezinger** [comment]
openhab version: openHAB 2.4.0-1
using the raspberrymatic as ccu wiht version:  3.41.11.20190126

since you cannot configure anything i cannot provide a tracelog.

**Comment by MHerbst** [comment]
> since you cannot configure anything i cannot provide a tracelog.

Please follow the instructions here: https://www.openhab.org/addons/bindings/homematic/#debugging-and-tracing and use TRACE level. Then restart openHAB. 
The Homematic binding creates most channels dynamically based on the information it gets from the CCU for a specific device. With activated trace mode the binding will log the communication with the CCU.

**Comment by agroezinger** [comment]
here is the log: 
https://pastebin.com/raw/eG95hdHY

**Comment by agroezinger** [comment]
Today I found this in the openhab.log after another restart. I dont know if it helps but:
https://pastebin.com/raw/Rg7mkdmm

**Comment by MHerbst** [comment]
Are you sure that the loglevel was set to TRACE? There are normally a lot more messages in the openhab.log. Or did you only look into the event.log? Which statement did you use to enable the TRACE?
Generally I can see quite a lot of status changes but the HMIP-PS gets online.

I have tried to find information about the data points provided by this device but did not find it in the manual :-(. 

You can ignore the messages regarding the names. This happens if a datapoint name (especially system variable) contain characters that are not supported as OH channel names.

**Comment by agroezinger** [comment]
Somehow I found the solution to the "bug"?

Finally i've figured out and got my HmIP-PS working.

I am using the configuration files for my setup. On my first tries I configured the thing like this:
> 	Thing   **HmIP-PS**         000218A9A6XXXX	    "Wohnzimmer Ecklampe"           @ "Wohnzimmer"

but doing it like this I ran into the problem, that no channels were found. Somehow I've had to configure the thing like this:
> 	Thing   **HMIP-PS**         000218A9A6XXXX	    "Wohnzimmer Ecklampe"           @ "Wohnzimmer"

The Thing identifier has to be written in UPPERCASE all the way, the m hast not to be lowercase like all my other Homematic things.

Can you please reply if this works for you too?

**Comment by MHerbst** [comment]
I have got some HM-IP devices but no HmIP-PS. I have created my things with the help of PaperUI and auto-detection. 
I have checked the thing definitions that are stored in the JSONDB and all Hm-IP Devices are written in mixed-case, e.g. "HmIP-PDT"
But it seems that eq3 has changed the naming scheme for the IP devices. In the download section of the website the older IP devices are written in mixed cased and the HMIP-PS device seems the be the first one that is named in all upper-case.
Therefore I think that writing the name in upper-case seems to be the solution.

@gerrieg If the thing names are case-sensitive, we should add a hint to the documentation. WDYT?

**Comment by agroezinger** [comment]
@MHerbst where do I find the JSONDB?

**Comment by MHerbst** [comment]
That depends on your installation. I have a manual installation on a Raspberry and it is located under <instaldir>/userdata/jsondb (I think you can simply search for jsondb). The automatically created things are stored in org.eclipse.smarthome.core.thing.Things.json.

**Comment by gerrieg** [comment]
> If the thing names are case-sensitive, we should add a hint to the documentation. 
yes, thanks you for the info. Will update the docs with my next PR
